modules/custom/activity_creator/js/notifications.js

```diff
@@ -4,30 +4,36 @@
  */
 
 (function ($) {
-    /**
-     * Behaviors.
-     */
-    Drupal.behaviors.notificationUpdate = {
-        attach: function (context, settings) {
-            // TODO Implement this with a no-js fallback.
-            var notification_count = $('.notification-bell .badge');
+  /**
+   * Notification centre bell update behavior.
+   */
+  Drupal.behaviors.notificationUpdate = {
+    attach: function (context, settings) {
 
-          if (notification_count.val() != "0") {
-              $('.notification-bell').click(function(e) {
-                  $.ajax({
-                      method: 'POST',
-                      url: '/ajax/notifications-mark-as-read',
-                      data: { },
-                      success: function(result) {
-                          // Update the notification bell.
-                          var remaining_notifications = result['remaining_notifications'];
+      $('.notification-bell', context).once('notificationUpdate').click(function(e) {
+        var notification_count = $('.notification-bell .badge');
 
-                          notification_count.html(remaining_notifications);
-                        $('.notification-bell.mobile .badge').html(remaining_notifications);
-                      }
-                    });
-              });
-          }
+        // We won't proceed if the notification count is 0 or the dropdown is
+        // open.
+        if (!notification_count.html() || notification_count.html() === "0" || $(this).hasClass('open')) {
```

Comment:
We could use the [jQuery data](https://api.jquery.com/data/) to store the count instead of reading the DOM which is an expensive operation.

Reply:
Added a todo as this would require more refactor work. Only implementing `data()` would be partial, as ideally we use `data-*` for keeping track of "data changes".